### PR TITLE
Feat:#219 enemy damage pop up

### PR DESCRIPTION
**Pull request by SalaheddineZ**
## Purpose of this PR:
Enemy has a damage pop up now.
## How to test:
Attack melee enemy in any scene, the damage should pop up

### links: 
[Ticket](https://github.com/SillyBusinessInc/SillyBusinessGame/issues/219)

### CODE REVIEW

File: Assets/Global/Scripts/Enemies/damagePopUp.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        if (duration <= activeDuration)
+        {
+            ResetColor();
+            textMesh.SetText("");
+            activeDuration = 0;
+            showDamage = 0;
+        }
+        else
+        {
+            Debug.Log(activeDuration);
```
Comment (PimMiii) [inline Assets/Global/Scripts/Enemies/damagePopUp.cs:28]: can be deleted

File: Assets/Global/Scripts/Enemies/damagePopUp.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+using TMPro;
+using UnityEngine;
+
+public class damagePopUp : MonoBehaviour
+{
+    private TextMeshPro textMesh;
+    private float showDamage = 0;
+    public float duration;
+    public float activeDuration;
+    public Color defaultColor;
```
Comment (PimMiii) [inline Assets/Global/Scripts/Enemies/damagePopUp.cs:10]: can be private / hidden in editor. Setting it in editor doesnt change anything, as the color set as Vertex Color in the text settings override it (might be the expected/preferred behavior anyways).

**Review by PimMiii — CHANGES_REQUESTED** [review]
couple very minor changes, mostly small cleanup/consistency tweaks.

File: Assets/Global/Scripts/Enemies/Base/EnemyBase.cs [hunk 3]
```diff
@@ -119,7 +120,9 @@ public abstract class EnemyBase : MonoBehaviour
         [Range(0f, 5f)]
         public float knockbackStunTime = 0.5f;
 
-        public GameObject HealthBarPrefab;
+        public GameObject healthBarPrefab;
+        public damagePopUp damagePopUp;
+        public float increasePopUpHeight;
```
[the remark this answers is not in the document]
Comment (SalaheddineZ) [reply Assets/Global/Scripts/Enemies/Base/EnemyBase.cs:125]: removed it

[the remark this answers is not in the document]
Comment (SalaheddineZ) [reply Assets/Global/Scripts/Enemies/Base/EnemyBase.cs:125]: yup removed

File: Assets/Global/Scripts/Enemies/damagePopUp.cs (hunk 2 above)
Comment (SalaheddineZ) [reply Assets/Global/Scripts/Enemies/damagePopUp.cs:10]: changed activeduration to private

File: Assets/Global/Scripts/Enemies/damagePopUp.cs (hunk 1 above)
Comment (SalaheddineZ) [reply Assets/Global/Scripts/Enemies/damagePopUp.cs:28]: yup did it

File: Assets/Global/Scripts/Enemies/damagePopUp.cs (hunk 2 above)
Comment (PimMiii) [reply Assets/Global/Scripts/Enemies/damagePopUp.cs:10]: think it was specifically, about the defaultColor.
image

**Review by PimMiii — CHANGES_REQUESTED** [review]
One last thing

**Review by PimMiii — APPROVED** [review]

**Review by WantToBeeMe — APPROVED** [review]

**Review by WantToBeeMe — APPROVED** [review]

**Pull request merged**